spec/javascripts/components/story/expanded_story/expanded_story_attachments_spec.js

```diff
@@ -40,11 +41,65 @@ describe('<ExpandedStoryAttachments />', () => {
         story={story}
         onAdd={onAdd}
         onDelete={onDelete}
+        disabled={false}
       />
     );
     const AttachmentsList = wrapper.find('AttachmentsList');
 
     expect(AttachmentsList).toExist();
     expect(AttachmentsList.prop('files')).toEqual(story._editing.documents)
   });
+
+  describe('when enabled', () => {
+    it('renders dropzone', () => {
+      const story = { _editing: { documents: [] } };
+
+      const wrapper = mount(
+        <ExpandedStoryAttachments
+          story={story}
+          onAdd={onAdd}
+          onDelete={onDelete}
+          disabled={false}
+        />
+      );
+
+      expect(wrapper.exists('Dropzone')).toBe(true);
+    });
+  });
+
+  describe('when disabled', () => {    
+    it('does not render dropzone', () => {
+      const documents = [{id: 1}];
+      const story = { documents, _editing: { documents } };
+
+      const wrapper = mount(
+        <ExpandedStoryAttachments
+          story={story}
+          onAdd={onAdd}
+          onDelete={onDelete}
+          disabled={true}
+        />
+      );
+
+      expect(wrapper.exists('Dropzone')).toBe(false);
+    });
+
+    describe('when there are no attachments', () => {
+      it('renders nothing', () => {
+        const documents = [];
+        const story = { documents, _editing: { documents } };
+
+        const wrapper = mount(
+          <ExpandedStoryAttachments
+            story={story}
+            onAdd={onAdd}
+            onDelete={onDelete}
+            disabled={true}
```

Comment:
Can we use `false` here to show that it won't render when there are no attachments even if it's enabled?

Reply:
I think we can't do that because if its enabled the story is not read-only, so it should render the attachments section and the upload button, right ?

Reply:
Can we change the description of the context (https://github.com/Codeminer42/cm42-central/pull/540/files/a17c6c997a68ee473a7678004bc95029d26abd58#diff-92c66b10be3e6fd942203618c2216a8dR87) then to `'when it is disabled and there are no attachments'`? It'll make it clearer when this condition happens.

Reply:
nevermind, it's right!